### PR TITLE
Multiple changes to furnaces and related recipes, as well as the recipe book.

### PR DESCRIPTION
**Pull request by EliteMasterEric**
* `FurnaceBlockEntity -> AbstractFurnaceBlockEntity` (since it is now the abstract parent of three furnace BlockEntities)
* `boc -> FurnaceBlockEntity`
* Additional function names for `SmokerBlockEntity`
* Additional function names for `BlastFurnaceBlockEntity`
* `FurnaceBlock -> AbstractFurnaceBlock` (now an abstract parent)
* `biw -> FurnaceBlock`
* Additional function names for `BlastFurnaceBlock`
* Additional function names for `SmokerBlock`
* `RecipeBookFurnaceGui -> AbstractFurnaceRecipeBookGui` (now an abstract parent, plus a bad naming scheme.)
* `cvz -> BlastFurnaceRecipeBookGui`
* `cwj -> FurnaceRecipeBookGui` (again, it should be most specific to least. Furnace -> Recipe Book -> Gui)
* `cwk -> SmokerRecipeBookGui`
* Additional field names for CraftingTableGui
* Additional field names for RecipeBookGui
* `crb -> CraftableFilterButtonWidget` (controls whether to show all recipes, or filter to just those that are craftable with the current inventory)
* Additional field names for `AbstractFurnaceContainer`
* Additional function names for `BlastFurnaceContainer`
* Additional function names for `FurnaceContainer`
* Additional function names for `SmokerContainer`
* Additional function name `incrementStat` for `EntityPlayer` (used to increment the stat that tracks how many times furnaces has been used)
* MANY fields added for `RecipeBook`

### CODE REVIEW

File: mappings/net/minecraft/container/AbstractFurnaceContainer.mapping [hunk 1]
```diff
@@ -13,9 +13,14 @@ CLASS asx net/minecraft/container/AbstractFurnaceContainer
...
-		ARG 1 player
 	METHOD d clearCraftingSlots ()V
 	METHOD e getCraftingResultSlotIndex ()I
 	METHOD f getCraftingWidth ()I
 	METHOD g getCrafitngHeight ()I
+	METHOD b transferSlot (Larb;I)Lawo;
+		ARG 1 player
+	METHOD b isFuel (Lawo;)Z
+		ARG 1 itemStack
+	METHOD a canCook (Lawo;)Z
```
Comment (asiekierka) [inline mappings/net/minecraft/container/AbstractFurnaceContainer.mapping:24]: canCook or canSmelt?

File: mappings/net/minecraft/block/FurnaceBlock.mapping [hunk 2]
```diff
@@ -1,43 +1,14 @@
...
-	METHOD a applyRotation (Lbpm;Lblb;)Lbpm;
-		ARG 1 state
-		ARG 2 rotation
-	METHOD a appendProperties (Lbpn$a;)V
-		ARG 1 builder
-	METHOD b hasComparatorOutput (Lbpm;)Z
-		ARG 1 state
-	METHOD c getRenderType (Lbpm;)Lbky;
-		ARG 1 state
+		ARG 4 rnd
```
Comment (asiekierka) [inline mappings/net/minecraft/block/FurnaceBlock.mapping:14]: rand

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/container/AbstractFurnaceContainer.mapping (hunk 1 above)
Comment (Prospector) [reply mappings/net/minecraft/container/AbstractFurnaceContainer.mapping:24]: canProcess IMO, as it's used for smelting, blasting, and smoking

File: mappings/net/minecraft/block/AbstractFurnaceBlock.mapping [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	FIELD a FACING Lbqh;
+	FIELD b ACTIVE Lbqe;
+	METHOD <init> (Lbgs$c;)V
+		ARG 1 settings
+	METHOD a getPlacementState (Laus;)Lbpm;
+		ARG 1 ctx
+	METHOD a openContainerGui (Lbbp;Let;Larb;)V
+		ARG 1 world
+		ARG 2 blockPos
+		ARG 3 playerentity
```
Comment (dblsaiko) [inline mappings/net/minecraft/block/AbstractFurnaceBlock.mapping:11]: ~~playerEntity~~, or `player`, actually

File: mappings/net/minecraft/block/BlastFurnaceBlock.mapping [hunk 4]
```diff
@@ -3,6 +3,11 @@ CLASS bgr net/minecraft/block/BlastFurnaceBlock
 		ARG 1 settings
 	METHOD a createBlockEntity (Lbbb;)Lbnq;
 		ARG 1 view
+	METHOD a openContainerGui (Lbbp;Let;Larb;)V
+		ARG 1 world
+		ARG 2 blockPos
+		ARG 3 playerentity
```
Comment (dblsaiko) [inline mappings/net/minecraft/block/BlastFurnaceBlock.mapping:9]: player

File: mappings/net/minecraft/block/FurnaceBlock.mapping [hunk 5]
```diff
@@ -1,43 +1,14 @@
...
-		ARG 4 placer
-	METHOD a getLuminance (Lbpm;)I
-		ARG 1 state
-	METHOD a getComparatorOutput (Lbpm;Lbbp;Let;)I
-		ARG 1 state
-		ARG 2 world
-		ARG 3 pos
-	METHOD a activate (Lbpm;Lbbp;Let;Larb;Lafo;Ley;FFF)Z
+		ARG 2 blockPos
+		ARG 3 playerentity
```
Comment (dblsaiko) [inline mappings/net/minecraft/block/FurnaceBlock.mapping:9]: player

File: mappings/net/minecraft/block/SmokerBlock.mapping [hunk 6]
```diff
@@ -3,6 +3,11 @@ CLASS blp net/minecraft/block/SmokerBlock
 		ARG 1 settings
 	METHOD a createBlockEntity (Lbbb;)Lbnq;
 		ARG 1 view
+	METHOD a openContainerGui (Lbbp;Let;Larb;)V
+		ARG 1 world
+		ARG 2 blockPos
+		ARG 3 playerentity
```
Comment (dblsaiko) [inline mappings/net/minecraft/block/SmokerBlock.mapping:9]: player

File: mappings/net/minecraft/client/gui/container/FurnaceRecipeBookGui.mapping [hunk 7]
```diff
@@ -0,0 +1,9 @@
+CLASS cwj net/minecraft/client/gui/container/FurnaceRecipeBookGui
+	METHOD a setFurnaceFilteringCraftable (Z)V
+		ARG 1 furnaceFilteringCraftable
+	METHOD b isFurnaceFilteringCraftable ()Z
+	METHOD d isFurnaceGuiOpen ()Z
+	METHOD d setFurnaceGuiOpen (Z)V
+		ARG 1 furnaceGuiOopen
```
Comment (dblsaiko) [inline mappings/net/minecraft/client/gui/container/FurnaceRecipeBookGui.mapping:7]: `furnaceGuiOopen` -> `furnaceGuiOpen`

**Review by dblsaiko — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/block/FurnaceBlock.mapping (hunk 2 above)
Comment (kashike) [reply mappings/net/minecraft/block/FurnaceBlock.mapping:14]: `random`

File: mappings/net/minecraft/container/AbstractFurnaceContainer.mapping (hunk 1 above)
Comment (asiekierka) [reply mappings/net/minecraft/container/AbstractFurnaceContainer.mapping:24]: canProcess.

File: mappings/net/minecraft/block/FurnaceBlock.mapping (hunk 2 above)
Comment (asiekierka) [reply mappings/net/minecraft/block/FurnaceBlock.mapping:14]: Sure

**Comment by asiekierka** [comment]
Also note that this will conflict highly with https://github.com/FabricMC/yarn/pull/289

File: mappings/net/minecraft/block/entity/AbstractFurnaceBlockEntity.mapping [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+		ARG 1 slot
+	METHOD a setInvStack (ILawo;)V
+		ARG 1 slot
+	METHOD a canInsertInvStack (ILawo;Ley;)Z
+		ARG 1 slot
+		ARG 2 stack
+	METHOD a (Lafi;)Z
+		ARG 0 inventory
+	METHOD a canPlayerUseInv (Larb;)Z
+		ARG 1 playerEntity
```
Comment (vladyslav-vasiliev) [inline mappings/net/minecraft/block/entity/AbstractFurnaceBlockEntity.mapping:29]: player

File: mappings/net/minecraft/block/entity/BlastFurnaceBlockEntity.mapping [hunk 9]
```diff
@@ -2,4 +2,10 @@ CLASS bnp net/minecraft/block/entity/BlastFurnaceBlockEntity
 	METHOD a createContainer (Lara;Larb;)Lasw;
 		ARG 1 playerInv
 	METHOD a canPlayerUseInv (Larb;)Z
+		ARG 1 playerEntity
```
Comment (vladyslav-vasiliev) [inline mappings/net/minecraft/block/entity/BlastFurnaceBlockEntity.mapping:5]: player

File: mappings/net/minecraft/block/entity/FurnaceBlockEntity.mapping [hunk 10]
```diff
@@ -1,24 +1,13 @@
...
-		ARG 1 slot
-	METHOD a canInsertInvStack (ILawo;Ley;)Z
-		ARG 1 slot
-		ARG 2 stack
+CLASS boc net/minecraft/block/entity/FurnaceBlockEntity
+	METHOD a createContainer (Lara;Larb;)Lasw;
+		ARG 1 playerInventory
+		ARG 2 playerEntity
 	METHOD a canPlayerUseInv (Larb;)Z
+		ARG 1 playerEntity
```
Comment (vladyslav-vasiliev) [inline mappings/net/minecraft/block/entity/FurnaceBlockEntity.mapping:6]: player

File: mappings/net/minecraft/block/entity/SmokerBlockEntity.mapping [hunk 11]
```diff
@@ -2,4 +2,10 @@ CLASS boo net/minecraft/block/entity/SmokerBlockEntity
 	METHOD a createContainer (Lara;Larb;)Lasw;
 		ARG 1 playerInv
 	METHOD a canPlayerUseInv (Larb;)Z
+		ARG 1 playerEntity
```
Comment (vladyslav-vasiliev) [inline mappings/net/minecraft/block/entity/SmokerBlockEntity.mapping:5]: player

**Comment by EliteMasterEric** [comment]
Closed because CodeWarrior apparently has no day job and can spend all his time resolving his pull requests and working on mapping areas other people are already mapping.

**Comment by jamierocks** [comment]
> Closed because CodeWarrior apparently has no day job and can spend all his time resolving his pull requests and working on mapping areas other people are already mapping.

Please try to remain civil. To be fair, CodeWarror's PR pre-dates yours - however, there is likely some stuff here that was not covered by CodeWarrior's PR.

We appreciate contributions from everyone, even if they don't end up getting merged / requiring changes.

**Pull request closed without merging**